Makefile

```diff
@@ -0,0 +1,157 @@
+mkfile_dir := $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
+
+##### Variables used
+# (can be overriden using command line args to make)
+# Paths used in build.
+BUILD_DIR:=$(abspath ${mkfile_dir}/../build)
+GECKO_SOURCE_DIR=$(abspath ${mkfile_dir}/../gecko-dev)
+GAIA_SOURCE_DIR=$(abspath ${mkfile_dir}/../gaia)
```

Comment:
Do you think we could default to dirs inside this work tree for gaia and gecko, and add `geck-dev` and `gaia` to `.gitignore`? It would be less confusing for external contributor.